### PR TITLE
created a GET /api/articles(topic query) endpoint

### PR DESCRIPTION
**Pull request by OzRikT95**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -50,28 +48,44 @@ describe("GET /api/articles/:article_id", () => {
...
-  test("responds with all articles, in descending order by date", async () => {
-    const { body } = await request(app).get("/api/articles").expect(200);
-    expect(body.articles).toBeSortedBy("created_at", { descending: true });
+  describe("GET /api/articles (topic query)", () => {
+    test("repond with array of articles filtered by topic", async () => {
+      const { body } = await request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200);
+      expect(body.articles).toBeInstanceOf(Array);
+    });
```
Comment (Harrison365) [inline __tests__/app.test.js:82]: Nice 👍 but we want to check that we get mitch articles back here.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -50,28 +48,44 @@ describe("GET /api/articles/:article_id", () => {
...
+    test("should repsond with err when no invalid topic ", async () => {
+      const { body } = await request(app)
+        .get("/api/articles?topic=mit")
+        .expect(404);
+      expect(body.msg).toBe("not found");
+    });
```
Comment (Harrison365) [inline __tests__/app.test.js:88]: Good 👍 
We need one more test where we ask for a topic which exists (200) but doesn't have any articles and returns an empty array.

File: model.js [hunk 3]
```diff
@@ -15,12 +16,20 @@ function getArticleById(articleId) {
...
-      return rows;
-    });
+function getArticles(topic) {
+  let query = "SELECT * FROM articles";
+  const topics = [];
+  if (topic) {
+    query += " WHERE topic = $1";
+    topics.push(topic);
+  }
+  query += " ORDER BY created_at DESC";
```
Comment (Harrison365) [inline model.js:26]: Nice 👍

File: model.js [hunk 4]
```diff
@@ -15,12 +16,20 @@ function getArticleById(articleId) {
...
+  return db.query(query, topics).then(({ rows }) => {
+    if (rows.length === 0) {
+      throw { status: 404, msg: "not found" };
+    }
+    return rows;
```
Comment (Harrison365) [inline model.js:31]: This will need to change for the extra test. You may need to do a separate check topic exists query.
